server/chat-commands/core.ts

```diff
@@ -733,10 +784,10 @@ export const commands: ChatCommands = {
 			for (const setting in user.settings) {
 				if (setting in raw) {
 					if (setting === 'blockPMs' &&
-						Users.Auth.isAuthLevel(raw[setting])) {
+						Users.Auth.isAuthLevel(raw[setting].all)) {
 						settings[setting] = raw[setting];
 					} else {
-						settings[setting as keyof UserSettings] = !!raw[setting];
+						settings[setting as keyof UserSettings] = typeof raw[setting] === 'object' ? raw[setting] : !!raw[setting];
```

Comment:
this does not work. this is the exact same problem. You are not guarding against bad input here.

Reply:
`{blockPMs: {foo: 'bar'}}` will pass through here.

Reply:
the question then, though, is how do you compare it? i'm planning to make a few more things an object and hardcoding just isn't practical.

Reply:
Actually, on testing, bad input doesn't crash or anything - maybe it'd be better to validate in their respective setting checkers?

Reply:
like, in canPM, if blockPMs.all isn't in Config.groupsranking and it's not a boolean, why not just clear it?

Reply:
I would rather validate on input than on use. You would have to set the type to `unknown` if you want to validate on use, and TypeScript basically makes it as unergonomic as possible. (TypeScript doesn't even let you use `a?.b` on things with type `unknown`!)

Reply:
So.... hardcoding it is, then?

Reply:
The alternative to "hardcoding" is creating some sort of generic typesafe JSON schema based validator which is the equivalent of using a sledgehammer to open a walnut (and specifying such schema for a basic type like settings would likely be both longer and more complicated for people to understand).

Reply:
Yeah, fair. 
This way at least we get to tailor it more specifically to each setting.